docker_cli/_docker_mamba.py

```diff
@@ -0,0 +1,235 @@
+    return textwrap.dedent(
+        """
+        ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+        USER $MAMBA_USER
+        """
+    ).strip()
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    packages: Iterable[str],
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+@overload
+def _mamba_spec_command(
+    command: str,
+    *,
+    channels: Optional[Iterable[str]],
+    specfile: os.PathLike[str] | str,
+    env_name: Optional[str] = ...,
+) -> str:
+    ...
+
+
+def _mamba_spec_command(
+    command, *, channels, specfile=None, packages=None, env_name=None
+) -> str:
+    if (specfile is not None) and (packages is not None):
+        raise ValueError("Specfile and packages both given. Use only one.")
+    if command not in ["install", "create"]:
+        raise ValueError(
+            "Improper value given for micromamba specs installation. "
+            'Please use one of "install" or "create".'
+        )
+
+    # If a name was given for the environment, an argument will be added to all
+    # micromamba installation commands.
+    if env_name is None:
+        name_arg: str = ""
+    else:
+        name_arg = f" -n {env_name}"
+
+    # Put together the channels argument
+    if channels is not None:
+        channels_arg = f" -c {shlex.join(channels)} --override-channels"
+
+    # If a specfile is given, give the instructions for copying and installing a
+    # specfile.
+    if specfile is not None:
+        install_command = textwrap.dedent(
+            f"""
+            COPY {specfile} /tmp/spec-file.txt
+            RUN micromamba {command}{name_arg}{channels_arg} -y -f /tmp/spec-file.txt \\
+             && rm /tmp/spec-file.txt \\
+        """
+        ).strip()
+    # Otherwise packages were given, so give the instructions for installing the
+    # packages.
+    else:
+        packages_str = shlex.join(packages)
+        install_command = textwrap.dedent(
+            f"""
+            RUN micromamba {command}{name_arg}{channels_arg} -y {packages_str} \\
+             && micromamba update --all \\
+        """
+        ).strip()
+
+    # Assemble this command into a portion of a Dockerfile string.
+    cmd: str = (
+        "ARG MAMBA_DOCKERFILE_ACTIVATE=1\n\n"
+        + install_command
+        + "\n"
+        + " && micromamba clean --all --yes"
+    )
+
+    return cmd
+
+
+def _mamba_install_prefix():
+    return "FROM mambaorg/micromamba:1.3.1 as micromamba"
+
+
+def _mamba_install_body(env_specfile: os.PathLike[str] | str = "spec-file.txt"):
+    # String variables to keep some of the COPY lines short
+    bin = "/usr/local/bin/"
+    activate_current_env = f"{bin}_activate_current_env.sh"
+    dockerfile_shell = f"{bin}_dockerfile_shell.sh"
+    entrypoint = f"{bin}_entrypoint.sh"
+    initialize_user_account = f"{bin}_dockerfile_initialize_user_accounts.sh"
+    setup_root_prefix = f"{bin}_dockerfile_setup_root_prefix.sh"
+
+    body: str = (
```

Comment:
We should include a link to the source that this Dockerfile snippet was adapted from. 

```suggestion
    # Adapted from https://micromamba-docker.readthedocs.io/en/latest/advanced_usage.html#adding-micromamba-to-an-existing-docker-image
    body: str = (
```